### PR TITLE
`to_chars`: migrating from custom switch to `__cpp_lib_to_chars`

### PR DESCRIPTION
**Pull request by fosterbrereton**
Previously the flag `ADOBE_HAS_TO_CHARS_FP` was derived from the existence (or lack) of a macOS-dependent preprocessor macro, which turned out to be flaky for the purposes of deciding if `to_chars` is available for use or not.

This PR instead switches the check to use the more standard `defined(__cpp_lib_to_chars) >= 201611L`. All checks of this switch have also been moved to source files to eliminate ODR violations. I commandeered the defunct `string.cpp` file to be the home of the `to_string` implementation.

### CODE REVIEW

**Comment by fosterbrereton** [comment]
Tests are _finally_ passing on all platforms. @sean-parent and @camio this one is ready for review.

File: vcpkg.json [hunk 1]
```diff
@@ -1,39 +1,21 @@
 {
     "name": "adobe-source-libraries",
     "version": "1.0.45",
```
Comment (sean-parent) [inline vcpkg.json:3]: Does that version number need to be updated when we do a release? Is that document?

Comment (fosterbrereton) [reply vcpkg.json:3]: I have no idea. I took that line verbatim from [what's in `main`](https://github.com/stlab/adobe_source_libraries/blob/main/vcpkg.json#L3) today. I'd ask @camio for details there.

**Review by sean-parent — APPROVED** [review]
LGTM - question about updating the vcpkg version but that is unrelated to this PR.

**Pull request merged**